test/python/circuit/library/test_multipliers.py

```diff
@@ -0,0 +1,123 @@
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test multiplier circuits."""
+
+import unittest
+import numpy as np
+from ddt import ddt, data, unpack
+
+from qiskit.test.base import QiskitTestCase
+from qiskit.circuit import QuantumCircuit
+from qiskit.quantum_info import Statevector
+from qiskit.circuit.library import (
+    RGQFTMultiplier,
+    HRSCumulativeMultiplier,
+    CDKMRippleCarryAdder,
+    DraperQFTAdder,
+    VBERippleCarryAdder,
+)
+
+
+@ddt
+class TestMultiplier(QiskitTestCase):
+    """Test the multiplier circuits."""
+
+    def assertMultiplicationIsCorrect(
+        self, num_state_qubits: int, num_result_qubits: int, multiplier: QuantumCircuit
+    ):
+        """Assert that multiplier correctly implements the product.
+
+        Args:
+            num_state_qubits: The number of bits in the numbers that are multiplied.
+            num_result_qubits: The number of qubits to limit the output to with modulo.
+            multiplier: The circuit performing the multiplication of two numbers with
+                ``num_state_qubits`` bits.
+        """
+        circuit = QuantumCircuit(*multiplier.qregs)
+
+        # create equal superposition
+        circuit.h(range(2 * num_state_qubits))
+
+        # apply multiplier circuit
+        circuit.compose(multiplier, inplace=True)
+
+        # obtain the statevector and the probabilities, we don't trace out the ancilla qubits
+        # as we verify that all ancilla qubits have been uncomputed to state 0 again
+        statevector = Statevector(circuit)
+        probabilities = statevector.probabilities()
+        pad = "0" * circuit.num_ancillas  # state of the ancillas
+
+        # compute the expected results
+        expectations = np.zeros_like(probabilities)
+        num_bits_product = num_state_qubits * 2
+        # iterate over all possible inputs
+        for x in range(2 ** num_state_qubits):
+            for y in range(2 ** num_state_qubits):
+                # compute the product
+                product = x * y % (2 ** num_result_qubits)
+                # compute correct index in statevector
+                bin_x = bin(x)[2:].zfill(num_state_qubits)
+                bin_y = bin(y)[2:].zfill(num_state_qubits)
+                bin_res = bin(product)[2:].zfill(num_bits_product)
+                bin_index = pad + bin_res + bin_y + bin_x
+                index = int(bin_index, 2)
+                expectations[index] += 1 / 2 ** (2 * num_state_qubits)
+        np.testing.assert_array_almost_equal(expectations, probabilities)
+
+    @data(
+        (3, RGQFTMultiplier),
```

Comment:
Did you also check corner cases like 0 or 1 number of state qubits or result qubits? 🙂

Reply:
I just tried running the following additional test cases locally:
```python
(RGQFTMultiplier, 0, 0),
(HRSCumulativeMultiplier, 0, 0),
(RGQFTMultiplier, 0, 1),
(HRSCumulativeMultiplier, 0, 1),
(RGQFTMultiplier, 1, 0),
(HRSCumulativeMultiplier, 1, 0),
```
All of them raised an error as expected. Would it be fine if I include these test cases as well? Asking just because there are quite a few already and I am not sure if that is too many as it is 😄

Reply:
Yeah it would be nice to add a few since they are running really fast.

Reply:
Thanks for clarifying this, I have added the additional test cases 👍